app/controllers/proposals_controller.rb

```diff
@@ -1,21 +1,24 @@
 class ProposalsController < ApplicationController
-  before_filter :require_event, except: :index
-  before_filter :require_user
-  before_filter :require_proposal, except: [ :index, :create, :new, :parse_edit_field ]
-  before_filter :require_invite_or_speaker, only: [:show]
+  before_action :require_event, except: :index
+  before_action :require_user
+  before_action :require_proposal, except: [ :index, :create, :new, :parse_edit_field ]
+  before_action :require_invite_or_speaker, only: [:show]
   skip_before_action :require_invite_or_speaker, only: [:destroy]
 
-  before_filter :require_speaker, only: [:edit, :update]
-  before_filter :require_waitlisted_or_accepted_state, only: [:confirm]
+  before_action :require_speaker, only: [:edit, :update]
+  before_action :require_waitlisted_or_accepted_state, only: [:confirm]
 
   decorates_assigned :proposal
 
   def index
-    proposals = current_user.proposals.decorate.to_a.group_by(&:event)
+    proposals = current_user.proposals.decorate.to_a.group_by {|p| p.event.decorate}
+    invitations = current_user.pending_invitations.decorate.to_a.group_by {|inv| inv.proposal.event.decorate}
+    events = (proposals.keys | invitations.keys).uniq
```

Comment:
Yeah good idea. I could have polished this a bit more. The event decoration was happening in the view originally, and I think that's a better place for it in this case.